hnn_core/tests/test_compare_hnn.py

```diff
@@ -102,3 +113,24 @@ def test_compare_across_backends():
                         atol=1e-14)
         assert_array_equal(dpls_reduced_default[trial_idx].data['agg'],
                            dpls_reduced_joblib[trial_idx].data['agg'])
+
+
+def test_mpi_failure():
+    """Test that an MPI failure is handled and error messages pass through"""
+    pytest.importorskip("mpi4py", reason="mpi4py not available")
+
+    # this MPI paramter will cause a MPI job with more than one process to fail
+    environ["OMPI_MCA_btl"] = "self"
```

Comment:
what's the practical implication of this? when do users set this parameter?

Reply:
It's to induce a failure and make sure that the messages are logged. If there are other failures, then at least they will be logged.